### PR TITLE
[docs][YSQL][#2314] CREATE RULE, DROP RULE

### PR DESCRIPTION
**Pull request by jj-kim**
This closes issue #2314.

### CODE REVIEW

File: docs/content/latest/api/ysql/commands/ddl_create_rule.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+Basic example.
+
+```sql
+yugabyte=# CREATE TABLE t1(a int4, b int4);
+yugabyte=# CREATE TABLE t2(a int4, b int4);
+yugabyte=# CREATE RULE t1_to_t2 AS ON INSERT TO t1 DO INSTEAD
+             INSERT INTO t2 VALUES (new.a, new.b);
+yugabyte=# INSERT INTO t1 VALUES (3, 4);
+yugabyte=# SELECT * FROM t1;
```
Comment (ndeodhar) [inline docs/content/latest/api/ysql/commands/ddl_create_rule.md]: Lets also show the results here.

File: docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf [hunk 2]
```diff
@@ -339,6 +339,15 @@ timestamp ::= '<DateTime Literal>' ;
 connlimit ::= '<Integer Literal>';
 uid ::= '<Text Literal>';
 
+(* 'CREATE' 'RULE' *)
+create_rule = 'CREATE' [ 'OR' 'REPLACE' ] 'RULE' rule_name 'AS' 'ON' rule_event
+              'TO' table_name [ 'WHERE' condition ]
+              'DO' [ 'ALSO' | 'INSTEAD' ] ( 'NOTHING' | command { ';' command } ) ;
```
Comment (ndeodhar) [inline docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf]: You're missing parentheses here:
```suggestion
              'DO' [ 'ALSO' | 'INSTEAD' ] ( 'NOTHING' | command | '(' command { ';' command } ')'  ) ;
```

Comment (jj-kim) [reply docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf]: I think parentheses are unnecessary here.  Adding them not only doesn't change the diagram but also doesn't change the generated grammar.

Comment (jj-kim) [reply docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf]: You're right; resolved.

**Pull request merged**